src/NewMatcher/alignment_types.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+from typing import Dict, NamedTuple, List, Union
+from src.data_types import (
+    BGC_Module_Prediction,
+    NRP_Monomer,
+    BGC_Variant,
+    LogProb,
+    NRP_Variant,
+    NRP_Fragment)
+from dataclasses import dataclass
+from prettytable import PrettyTable
+from io import StringIO
+
+
+class AlignmentStep(NamedTuple):
+    bgc_module: Union[BGC_Module_Prediction, None]
+    nrp_monomer: Union[NRP_Monomer, None]
+    score: LogProb
+    action: str
+
+    def to_dict(self) -> Dict[str, str]:
+        NA = '---'
+        return {'Gene': self.bgc_module.gene_id if self.bgc_module else NA,
+                'A-domain_idx': self.bgc_module.module_idx if self.bgc_module else NA,
+                'Modifications': ','.join(mod.name for mod in self.bgc_module.modifications)
```

Comment:
Well, I googled that the proper term here should be "Modifying enzymes" (or maybe "Modifying_domains" is closer to our terminology and still quite correct from biological point of view)